### PR TITLE
infoschema: refine error code for package `infoschema`

### PR DESCRIPTION
**Pull request by Deardrops**
### What problem does this PR solve? 

Not all the errors in infoschema package have error code.

Make some errors related to `CREATE USER / ALTER USER / DROP USER` statement compatibly with MySQL 5.7, e.g.

```
mysql> warnings;
Show warnings enabled.

mysql> create user if not exists 'newuser'@'localhost';
Query OK, 0 rows affected (0.00 sec)

mysql> create user if not exists 'newuser'@'localhost';
Query OK, 0 rows affected, 1 warning (0.00 sec)

Note (Code 3163): User 'newuser'@'localhost' already exists.
mysql> create user 'newuser'@'localhost';
ERROR 1396 (HY000): Operation CREATE USER failed for 'newuser'@'localhost'
```
### What is changed and how it works?

Add them, related PR pingcap/parser#644

### Check List 

Tests 

 - Integration test

Code changes

 - Has exported variable/fields change

Side effects

 - None

Related changes

 - None

Release note

 - None

### CODE REVIEW

File: executor/simple_test.go [hunk 1]
```diff
@@ -275,7 +274,7 @@ func (s *testSuite3) TestUser(c *C) {
 	result = tk.MustQuery(`SELECT Password FROM mysql.User WHERE User="test1" and Host="localhost"`)
 	result.Check(testkit.Rows(auth.EncodePassword("111")))
 	alterUserSQL = `ALTER USER IF EXISTS 'test2'@'localhost' IDENTIFIED BY '222', 'test_not_exist'@'localhost' IDENTIFIED BY '1';`
-	_, err = tk.Exec(alterUserSQL)
+	_, err := tk.Exec(alterUserSQL)
```
Comment (bb7133) [inline executor/simple_test.go]: Change it to `MustGetErrCode`?

File: executor/simple.go [hunk 2]
```diff
@@ -659,6 +653,12 @@ func (e *SimpleExec) executeCreateUser(ctx context.Context, s *ast.CreateUserStm
 		if s.IsCreateRole {
 			user = fmt.Sprintf(`('%s', '%s', '%s', 'Y')`, spec.User.Hostname, spec.User.Username, pwd)
 		}
+		if exists {
+			if !s.IfNotExists {
+				return infoschema.ErrUserAlreadyExists.GenWithStackByArgs(user)
```
Comment (bb7133) [inline executor/simple.go]:
```suggestion
				return infoschema.ErrUserAlreadyExists.GenWithStackByArgs(fmt.Sprintf(`'%s'@'%s'`), spec.User.Hostname, spec.User.Username)
```
How about changing the message like this?

File: executor/simple.go [hunk 3]
```diff
@@ -659,6 +653,12 @@ func (e *SimpleExec) executeCreateUser(ctx context.Context, s *ast.CreateUserStm
 		if s.IsCreateRole {
 			user = fmt.Sprintf(`('%s', '%s', '%s', 'Y')`, spec.User.Hostname, spec.User.Username, pwd)
 		}
+		if exists {
+			if !s.IfNotExists {
```
Comment (bb7133) [inline executor/simple.go]: I found MySQL used a different error type:

```
mysql> CREATE USER 'newuser'@'localhost';
Query OK, 0 rows affected (0.00 sec)

mysql> CREATE USER 'newuser'@'localhost';
ERROR 1396 (HY000): Operation CREATE USER failed for 'newuser'@'localhost'
```
PTAL, however, I think using `ErrUserAlreadyExists` is better.

**Comment by jackysp** [comment]
@Deardrops merge failed, PTAL

**Pull request merged**